source/com.microsoft.tfs.core/rest_core/com/microsoft/alm/client/TeeClientHandler.java

```diff
@@ -0,0 +1,371 @@
+        request.setFollowRedirects(true);
+
+        try {
+            request.setRequestHeader(VssHttpHeaders.ACCEPT, VssMediaTypes.APPLICATION_JSON_TYPE);
+            final int statusCode = httpClient.executeMethod(request);
+
+            if (HttpStatus.isSuccessFamily(statusCode)) {
+                return JsonHelper.deserializeResponce(request, ApiResourceLocationCollection.class);
+            } else {
+                throw new HttpException(HttpStatus.getStatusText(statusCode));
+            }
+        } catch (final Exception e) {
+            log.error(e.getMessage(), e);
+            setLastException(e);
+            return null;
+        } finally {
+            request.releaseConnection();
+        }
+    }
+
+    @Override
+    public <TEntity> VssRestRequest createRequest(
+        final HttpMethod method,
+        final UUID locationId,
+        final Map<String, Object> routeValues,
+        final ApiResourceVersion version,
+        final TEntity value,
+        final String contentMediaType,
+        final Map<String, String> queryParameters,
+        final String acceptMediaType) {
+
+        final URI target = createTarget(locationId, routeValues, queryParameters);
+        final String acceptType =
+            getMediaTypeWithQualityHeaderValue(acceptMediaType, negotiateRequestVersion(locationId, version));
+
+        final HttpMethodBase request;
+        if (shouldOverrideHttpMethod(method)) {
+            request = createHttpMethod(HttpMethod.POST, target);
+            request.setRequestHeader(VssHttpHeaders.HTTP_METHOD_OVERRIDE, method.toString());
+        } else {
+            request = createHttpMethod(method, target);
+        }
+
+        request.setRequestHeader(VssHttpHeaders.ACCEPT, acceptType);
+        request.setRequestHeader(VssHttpHeaders.TFS_VERSION, CoreVersionInfo.getVersion());
+
+        if (request instanceof EntityEnclosingMethod) {
+            final String contentType = getMediaTypeWithQualityHeaderValue(contentMediaType);
+            final EntityEnclosingMethod entityMethod = (EntityEnclosingMethod) request;
+
+            RequestEntity entity = null;
+
+            if (value != null && (value instanceof InputStream)) {
+                entity = new InputStreamRequestEntity((InputStream) value, contentMediaType);
+            } else {
+                final String content;
+                if (value != null) {
+                    content = JsonHelper.serializeRequestToString(value);
+                } else {
+                    content = StringUtil.EMPTY;
+                }
+
+                try {
+                    entity = new StringRequestEntity(content, contentType, StringUtil.UTF8_CHARSET);
+                } catch (final UnsupportedEncodingException e) {
+                    // Note that this exception cannot happen at this point
+                    log.error("Illegal content encoding found.", e); //$NON-NLS-1$
```

Comment:
If we are sure this exception can't happen at this point, why not throw it up as a runtime exception?  If we are here something's obviously wrong?